sympy/solvers/diophantine/diophantine.py

```diff
@@ -98,18 +97,17 @@ class DiophantineSolutionSet(set):
         {(t**2, t + 2)}
     """
 
-    def __init__(self, symbols_seq, parameters=None):
+    def __init__(self, symbols_seq, parameters):
```

Comment:
IMO, this is a public API change. Do we need a deprecation warning for this?

Reply:
@oscarbenjamin What do you say about this change and the overall PR?

Reply:
I think the PR looks good. We should resolve the merge conflicts and get it in.